swabbie-core/src/main/kotlin/com/netflix/spinnaker/swabbie/LockingService.kt

```diff
@@ -18,12 +18,31 @@ package com.netflix.spinnaker.swabbie
 
 import com.netflix.spinnaker.config.LockingConfigurationProperties
 import com.netflix.spinnaker.kork.lock.LockManager
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
 import java.time.Duration
+import java.util.concurrent.atomic.AtomicInteger
 
 class LockingService(
   private val lockManager: LockManager,
   private val lockingConfigurationProperties: LockingConfigurationProperties
 ) : LockManager by lockManager {
+  private val log: Logger = LoggerFactory.getLogger(javaClass)
+
+  val numLocksCurrentlyHeld = AtomicInteger(0)
+
   val swabbieMaxLockDuration: Duration
     get() = Duration.ofMillis(lockingConfigurationProperties.maximumLockDurationMillis!!)
+
+  val swabbieMaxConcurrentLocks: Int
+    get() = lockingConfigurationProperties.maxConcurrentLocks
+
+  @Synchronized override fun acquireLock(lockOptions: LockManager.LockOptions, onLockAcquiredCallback: Runnable): LockManager.AcquireLockResponse<Void> {
+    if (numLocksCurrentlyHeld.get() >= lockingConfigurationProperties.maxConcurrentLocks) {
+      log.warn("Too many locks are held " +
+          "(held=${numLocksCurrentlyHeld.get()}, wanted=${lockingConfigurationProperties.maxConcurrentLocks}).")
+    }
+    numLocksCurrentlyHeld.incrementAndGet()
```

Comment:
probably want to only increment this if the acquireLock is successful

Reply:
is there nothing explicitly calling releaseLock on these locks? (I could see that, assuming the model is acquire the lock then let it TTL away before the next instance can acquire it)

Reply:
No, from my sleuthing it seems they're being called within the redisLockManager. I might be able to override that also, but was having trouble. Because decrementing there would be better than in the `finally` of each function.